pkg/sbom/cyclonedx/marshal.go

```diff
@@ -185,7 +204,12 @@ type Package struct {
 
 func (e *Marshaler) marshalPackage(pkg Package, pkgs map[string]Package, components map[string]*core.Component,
 ) (*core.Component, error) {
-	if c, ok := components[pkg.ID]; ok {
+	// use a hash function instead of just the pkg.ID because applications can have multiple packages with the same name
+	pkgID, err := calcPkgID(e.hasher, pkg)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get %s package ID: %w", pkg.Name, err)
+	}
+	if c, ok := components[pkgID]; ok {
```

Comment:
Do you think we need this?
If I understand correctly, pkg.ID (this is the UUID from the julia file) is always unique.

Reply:
Sorry this was left over from an earlier version. Removed